src/views/StackView/screens.js

```diff
@@ -1,6 +1,46 @@
-import { Animated, View } from 'react-native';
+import React from 'react';
+import { Animated, View, UIManager, StyleSheet } from 'react-native';
 
-const ScreenContainer = View;
-const Screen = Animated.View;
+let ScreenContainer = View;
+let Screen = Animated.View;
+
+// If RNSScreen native component is available, instead of using plain RN Views
+// for ScreenContainer and Screen components, we can use native component
+// provided by react-native-screens lib. Note that we don't specify
+// react-native-screens as a dependency, but instead we check whether the library
+// is linked natively (we only `require` the lib if native components are installed)
+if (UIManager['RNSScreen']) {
+  // native screen components are available
+  const screens = require('react-native-screens');
```

Comment:
this won't work if `react-native-screens` isn't installed, even though we gate it with the UIManager conditional. users would need to use something like https://github.com/satya164/babel-plugin-optional-require or they will get an error in metro. we use a global for this in a couple situations in expo :\ but then that introduces the problem of needing to import that package for its global side effect before the global is depended on. maybe we should get optional-require in react-native upstream because this is going to increasingly become a problem as react-native becomes more modular.

Reply:
For now I can add react-native-screens as dependency. We will only check here if native code is linked or not. What do you think?

Reply:
That sounds reasonable to me